### PR TITLE
Symfony6対応

### PR DESCRIPTION
**Pull request by chihiro-adachi**
## 概要(Overview・Refs Issue)

#6069 のシステム要件・フレームワークの更新に対応

- PHP8.3対応
- Symfony6対応
- JSライブラリの更新

### アップデートするパッケージ

https://github.com/EC-CUBE/ec-cube/issues/6069 を参照

<details>

  - Removing symfony/security-guard (v5.4.19)
  - Removing symfony/inflector (v5.4.11)
  - Removing easycorp/easy-log-handler (v1.0.9)
  - Removing codeception/phpunit-wrapper (9.0.9)
  - Upgrading symfony/flex (v1.19.4 => v1.21.4)
  - Upgrading symfony/polyfill-mbstring (v1.27.0 => v1.28.0)
  - Upgrading symfony/polyfill-iconv (v1.27.0 => v1.28.0)
  - Upgrading zbateson/mb-wrapper (1.1.2 => 1.2.0)
  - Upgrading psr/http-message (1.0.1 => 1.1)
  - Upgrading psr/http-factory (1.0.1 => 1.0.2)
  - Upgrading guzzlehttp/psr7 (2.4.3 => 2.6.2)
  - Upgrading zbateson/stream-decorators (1.0.7 => 1.2.1)
  - Upgrading symfony/deprecation-contracts (v2.5.2 => v3.4.0)
  - Upgrading psr/http-client (1.0.1 => 1.0.3)
  - Upgrading guzzlehttp/promises (1.5.2 => 2.0.2)
  - Upgrading guzzlehttp/guzzle (7.5.0 => 7.8.1)
  - Upgrading symfony/polyfill-ctype (v1.27.0 => v1.28.0)
  - Upgrading symfony/yaml (v5.4.17 => v6.4.0)
  - Upgrading symfony/var-dumper (v5.4.19 => v6.4.2)
  - Upgrading symfony/finder (v5.4.17 => v6.4.0)
  - Upgrading symfony/event-dispatcher-contracts (v2.5.2 => v3.4.0)
  - Upgrading symfony/event-dispatcher (v5.4.19 => v6.4.2)
  - Upgrading symfony/css-selector (v5.4.17 => v6.4.0)
  - Upgrading symfony/polyfill-intl-normalizer (v1.27.0 => v1.28.0)
  - Upgrading symfony/polyfill-intl-grapheme (v1.27.0 => v1.28.0)
  - Upgrading symfony/string (v5.4.19 => v6.4.2)
  - Upgrading symfony/service-contracts (v2.5.2 => v3.4.1)
  - Upgrading symfony/console (v5.4.17 => v6.4.2)
  - Upgrading sebastian/diff (4.0.4 => 4.0.5)
  - Upgrading sebastian/recursion-context (4.0.4 => 4.0.5)
  - Upgrading nikic/php-parser (v4.15.2 => v4.18.0)
  - Installing psy/psysh (v0.12.0)
  - Upgrading sebastian/type (3.2.0 => 3.2.1)
  - Upgrading sebastian/global-state (5.0.5 => 5.0.6)
  - Upgrading sebastian/environment (5.1.4 => 5.1.5)
  - Upgrading theseer/tokenizer (1.2.1 => 1.2.2)
  - Upgrading sebastian/lines-of-code (1.0.3 => 1.0.4)
  - Upgrading sebastian/complexity (2.0.2 => 2.0.3)
  - Upgrading phpunit/php-code-coverage (9.2.23 => 9.2.30)
  - Upgrading myclabs/deep-copy (1.11.0 => 1.11.1)
  - Upgrading phpunit/phpunit (9.5.27 => 9.6.16)
  - Upgrading codeception/stub (4.0.2 => 4.1.2)
  - Upgrading codeception/codeception (4.2.2 => 5.0.13)
  - Upgrading codeception/lib-asserts (1.13.2 => 2.1.0)
  - Upgrading captbaritone/mailcatcher-codeception-module (2.4.1 => 3.0.0)
  - Upgrading codeception/module-asserts (1.3.1 => 3.0.0)
  - Installing masterminds/html5 (2.8.1)
  - Upgrading symfony/dom-crawler (v5.4.17 => v6.4.0)
  - Upgrading symfony/browser-kit (v5.4.11 => v6.4.0)
  - Installing codeception/lib-web (1.0.5)
  - Upgrading codeception/lib-innerbrowser (1.5.1 => 3.1.3)
  - Upgrading codeception/module-phpbrowser (1.0.3 => 3.0.1)
  - Upgrading softcreatr/jsonpath (0.7.5 => 0.8.3)
  - Upgrading justinrainbow/json-schema (5.2.12 => v5.2.13)
  - Installing codeception/lib-xml (1.0.2)
  - Upgrading codeception/module-rest (1.4.2 => 3.3.2)
  - Upgrading symfony/process (v5.4.11 => v6.4.2)
  - Upgrading php-webdriver/webdriver (1.13.1 => 1.15.1)
  - Upgrading codeception/module-webdriver (1.4.1 => 3.2.1)
  - Upgrading symfony/polyfill-php81 (v1.27.0 => v1.28.0)
  - Upgrading symfony/polyfill-php80 (v1.27.0 => v1.28.0)
  - Upgrading symfony/polyfill-php73 (v1.27.0 => v1.28.0)
  - Upgrading symfony/filesystem (v5.4.19 => v6.4.0)
  - Upgrading seld/signal-handler (2.0.1 => 2.0.2)
  - Upgrading seld/jsonlint (1.9.0 => 1.10.1)
  - Upgrading react/promise (v2.9.0 => v3.1.0)
  - Upgrading composer/pcre (3.1.0 => 3.1.1)
  - Upgrading composer/spdx-licenses (1.5.7 => 1.5.8)
  - Upgrading composer/semver (3.3.2 => 3.4.0)
  - Upgrading composer/class-map-generator (1.0.0 => 1.1.0)
  - Upgrading composer/ca-bundle (1.3.4 => 1.4.0)
  - Upgrading composer/composer (2.5.1 => 2.6.6)
  - Upgrading symfony/routing (v5.4.17 => v6.4.2)
  - Installing symfony/polyfill-php83 (v1.28.0)
  - Upgrading symfony/http-foundation (v5.4.20 => v6.4.2)
  - Upgrading symfony/error-handler (v5.4.19 => v6.4.0)
  - Upgrading symfony/http-kernel (v5.4.20 => v6.4.2)
  - Upgrading symfony/var-exporter (v5.4.17 => v6.4.2)
  - Upgrading symfony/dependency-injection (v5.4.20 => v6.4.2)
  - Upgrading symfony/config (v5.4.19 => v6.4.0)
  - Upgrading psr/cache (1.0.1 => 3.0.0)
  - Upgrading symfony/cache-contracts (v2.5.2 => v3.4.0)
  - Upgrading symfony/cache (v5.4.18 => v6.4.2)
  - Upgrading symfony/framework-bundle (v5.4.17 => v6.4.2)
  - Upgrading doctrine/event-manager (1.2.0 => 2.0.0)
  - Upgrading doctrine/persistence (2.5.6 => 3.2.0)
  - Upgrading symfony/doctrine-bridge (v5.4.17 => v6.4.2)
  - Upgrading doctrine/deprecations (v1.0.0 => 1.1.2)
  - Upgrading doctrine/dbal (3.5.2 => 3.8.0)
  - Upgrading doctrine/doctrine-bundle (2.8.2 => 2.11.1)
  - Upgrading symfony/polyfill-php72 (v1.27.0 => v1.28.0)
  - Upgrading doctrine/lexer (1.2.3 => 2.1.0)
  - Upgrading doctrine/inflector (2.0.6 => 2.0.9)
  - Upgrading doctrine/orm (2.14.0 => 2.17.4)
  - Upgrading doctrine/data-fixtures (1.6.3 => 1.7.0)
  - Upgrading doctrine/doctrine-fixtures-bundle (3.4.2 => 3.5.1)
  - Upgrading symfony/stopwatch (v5.4.13 => v6.4.0)
  - Upgrading doctrine/migrations (3.5.2 => 3.7.2)
  - Upgrading doctrine/doctrine-migrations-bundle (3.2.2 => 3.3.0)
  - Upgrading ezyang/htmlpurifier (v4.16.0 => v4.17.0)
  - Upgrading exercise/htmlpurifier-bundle (v3.1.0 => 4.1.2)
  - Upgrading fakerphp/faker (v1.21.0 => v1.23.1)
  - Upgrading symfony/options-resolver (v5.4.11 => v6.4.0)
  - Upgrading friendsofphp/php-cs-fixer (v3.13.2 => v3.48.0)
  - Upgrading twig/twig (v2.15.4 => v3.8.0)
  - Upgrading symfony/translation-contracts (v2.5.2 => v3.4.1)
  - Upgrading symfony/translation (v5.4.14 => v6.4.2)
  - Upgrading laminas/laminas-code (4.7.1 => 4.13.0)
  - Upgrading mobiledetect/mobiledetectlib (2.8.41 => 2.8.45)
  - Installing psr/clock (1.0.0)
  - Installing carbonphp/carbon-doctrine-types (2.1.0)
  - Upgrading nesbot/carbon (2.65.0 => 2.72.2)
  - Upgrading phpstan/phpstan (1.9.8 => 1.10.57)
  - Upgrading doctrine/annotations (1.14.2 => 2.0.1)
  - Upgrading sensio/framework-extra-bundle (v6.2.9 => v6.2.10)
  - Upgrading setasign/fpdi (v2.3.6 => v2.6.0)
  - Upgrading symfony/asset (v5.4.13 => v6.4.0)
  - Upgrading symfony/twig-bridge (v5.4.17 => v6.4.2)
  - Upgrading symfony/debug-bundle (v5.4.11 => v6.4.0)
  - Upgrading symfony/dotenv (v5.4.5 => v6.4.2)
  - Upgrading symfony/expression-language (v5.4.14 => v6.4.2)
  - Upgrading symfony/property-info (v5.4.19 => v6.4.0)
  - Upgrading symfony/property-access (v5.4.19 => v6.4.0)
  - Upgrading symfony/polyfill-intl-icu (v1.27.0 => v1.28.0)
  - Upgrading symfony/form (v5.4.17 => v6.4.1)
  - Upgrading symfony/lock (v5.4.15 => v6.4.2)
  - Upgrading symfony/polyfill-intl-idn (v1.27.0 => v1.28.0)
  - Upgrading symfony/mime (v5.4.17 => v6.4.0)
  - Upgrading egulias/email-validator (2.1.25 => 4.0.2)
  - Upgrading symfony/mailer (v5.4.17 => v6.4.2)
  - Upgrading symfony/maker-bundle (v1.43.0 => v1.52.0)
  - Upgrading monolog/monolog (2.8.0 => 2.9.2)
  - Upgrading symfony/monolog-bridge (v5.4.17 => v6.4.0)
  - Upgrading symfony/monolog-bundle (v3.8.0 => v3.10.0)
  - Upgrading symfony/phpunit-bridge (v5.4.17 => v6.4.2)
  - Upgrading friendsofphp/proxy-manager-lts (v1.0.13 => v1.0.16)
  - Upgrading symfony/proxy-manager-bridge (v5.4.6 => v6.4.0)
  - Upgrading symfony/rate-limiter (v5.4.17 => v6.4.2)
  - Upgrading symfony/password-hasher (v5.4.19 => v6.4.0)
  - Upgrading symfony/security-core (v5.4.19 => v6.4.0)
  - Upgrading symfony/security-http (v5.4.20 => v6.4.0)
  - Upgrading symfony/security-csrf (v5.4.19 => v6.4.0)
  - Installing symfony/clock (v6.4.2)
  - Upgrading symfony/security-bundle (v5.4.20 => v6.4.2)
  - Upgrading symfony/serializer (v5.4.17 => v6.4.2)
  - Upgrading symfony/templating (v5.4.11 => v6.4.0)
  - Upgrading symfony/validator (v5.4.17 => v6.4.2)
  - Upgrading symfony/twig-bundle (v5.4.17 => v6.4.0)
  - Upgrading symfony/web-profiler-bundle (v5.4.17 => v6.4.2)
  - Upgrading symfony/workflow (v5.4.17 => v6.4.2)
  - Upgrading tecnickcom/tcpdf (6.6.2 => 6.6.5)
  - Upgrading twig/extra-bundle (v3.5.0 => v3.8.0)
  - Upgrading symfony/intl (v5.4.15 => v6.4.2)
  - Upgrading twig/intl-extra (v3.5.0 => v3.8.0)
  - Upgrading phpoption/phpoption (1.9.0 => 1.9.2)
  - Upgrading graham-campbell/result-type (v1.1.0 => v1.1.2)

</details>

### 削除されるパッケージ

- symfony/security-guard (v5.4.19)
- symfony/inflector (v5.4.11)
- easycorp/easy-log-handler (v1.0.9)
  - devのログははprodと同等の出力に変更
- codeception/phpunit-wrapper (9.0.9)

### 主な変更点

- PasswordEncoderの廃止
  - PasswordHasherを利用するように変更
  - パスワードのハッシュアルゴリズムの自動更新にも対応
  - ログイン時にEC-CUBEのハッシュアルゴリズムから自動更新される https://github.com/EC-CUBE/ec-cube/pull/6073/commits/3e1d7b7d37fa0af3344f231d4b07087d1cbdbfa8
- Twig3対応
  - twigを2から3へアップデート。プラグインの拡張には影響しない https://github.com/EC-CUBE/ec-cube/pull/6073/commits/7863e8e69734f3c223a4f46e06ab76f34b5b5719
- Codeception5対応
  - 一部非互換のメソッドなどあり。コード差分参照してください。

### TODO

- Bootstrapの更新により、一部デザイン調整が必要 https://github.com/EC-CUBE/ec-cube/issues/6095
- プラグインのインストールに失敗するため、deny-test.ymlを一部コメントアウトしている https://github.com/EC-CUBE/ec-cube/issues/6096
- PHPUnit10は現状非対応。余力があれば調査する

## テスト（Test)

github actionsは一通りパス

## 相談（Discussion）

- unit testの実行時間が従来5分程度だったのが20分程度かかるようになっていた
- パスワードのハッシュ化の影響によるものであったため、[テスト時はハッシュ化のコストを最小限にするように修正](https://github.com/EC-CUBE/ec-cube/pull/6073/commits/1ee1283fef5303db5ff260abbb74df802883ba6a)し、10分程度に短縮はできたがもう少し改善したい

## マイナーバージョン互換性保持のための制限事項チェックリスト

- [ ] 既存機能の仕様変更はありません
- [x] フックポイントの呼び出しタイミングの変更はありません
- [x] フックポイントのパラメータの削除・データ型の変更はありません
- [x] twigファイルに渡しているパラメータの削除・データ型の変更はありません
- [ ] Serviceクラスの公開関数の、引数の削除・データ型の変更はありません
- [x] 入出力ファイル(CSVなど)のフォーマット変更はありません

## レビュワー確認項目

- [ ] 動作確認
- [ ] コードレビュー
- [ ] E2E/Unit テスト確認(テストの追加・変更が必要かどうか)
- [ ] 互換性が保持されているか
- [ ] セキュリティ上の問題がないか
  - [ ] 権限を超えた操作が可能にならないか
  - [ ] 不要なファイルアップロードがないか
  - [ ] 外部へ公開されるファイルや機能の追加ではないか
  - [ ] テンプレートでのエスケープ漏れがないか

### CODE REVIEW

**Comment by chihiro-adachi** [comment]
TODO
- UTの実行時間が20分以上かかる
- SameSiteCookieの挙動変更
- DoctrineのCache

File: src/Eccube/Common/EccubeConfig.php [hunk 1]
```diff
@@ -85,7 +76,7 @@ public function offsetGet($offset)
     #[\ReturnTypeWillChange]
     public function offsetSet($offset, $value)
     {
-        $this->set($offset, $value);
+        throw new \Exception();
```
Comment (nanasess) [inline src/Eccube/Common/EccubeConfig.php]: これが呼ばれるのはプログラミングのミスだと思うので、 LogickException で良いと思います
```suggestion
        throw new \LogicException();
```

File: src/Eccube/Security/Core/User/CustomerProvider.php [hunk 2]
```diff
@@ -86,4 +95,16 @@ public function supportsClass($class)
     {
         return Customer::class === $class || is_subclass_of($class, Customer::class);
     }
+
+    public function loadUserByIdentifier(string $identifier): UserInterface
+    {
+        // FIXME deprecated
+        return $this->loadUserByUsername($identifier);
```
Comment (nanasess) [inline src/Eccube/Security/Core/User/CustomerProvider.php]: `loadUserByUsername()` が deprecated なので、`loadUserByIdentifier` に処理を実装して、 loadUserByUsername() の中で loadUserByIdentifier を呼んだ方が良いと思います

File: src/Eccube/Security/Core/User/MemberProvider.php [hunk 3]
```diff
@@ -83,4 +92,16 @@ public function supportsClass($class)
     {
         return Member::class === $class || is_subclass_of($class, Member::class);
     }
+
+    public function loadUserByIdentifier(string $identifier): UserInterface
+    {
+        // FIXME deprecated
+        return $this->loadUserByUsername($identifier);
```
Comment (nanasess) [inline src/Eccube/Security/Core/User/MemberProvider.php]: `loadUserByUsername()` が deprecated なので、`loadUserByIdentifier` に処理を実装して、 loadUserByUsername() の中で loadUserByIdentifier を呼んだ方が良いと思います

File: src/Eccube/Command/GenerateProxyCommand.php [hunk 4]
```diff
@@ -50,12 +50,13 @@ protected function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         // アノテーションを読み込めるように設定.
-        AnnotationRegistry::registerAutoloadNamespace('Eccube\Annotation', __DIR__.'/../../../src');
+        // TODO doctrine/anntationsをv2へアップデート。影響がある場合は要調査。
```
Comment (nanasess) [inline src/Eccube/Command/GenerateProxyCommand.php]: ここのコメントが気になります

**Review by nanasess — CHANGES_REQUESTED** [review]

File: src/Eccube/Common/EccubeConfig.php (hunk 1 above)
Comment (chihiro-adachi) [reply src/Eccube/Common/EccubeConfig.php]: ありがとうございます、修正します。

File: src/Eccube/Security/Core/User/CustomerProvider.php (hunk 2 above)
Comment (chihiro-adachi) [reply src/Eccube/Security/Core/User/CustomerProvider.php]: たしかにそうですね。修正します。

File: src/Eccube/Security/Core/User/MemberProvider.php (hunk 3 above)
Comment (chihiro-adachi) [reply src/Eccube/Security/Core/User/MemberProvider.php]: たしかにそうですね。修正します。

File: src/Eccube/Command/GenerateProxyCommand.php (hunk 4 above)
Comment (chihiro-adachi) [reply src/Eccube/Command/GenerateProxyCommand.php]: @nanasess 
ありがとうございます。
doctrine/anntationsをv2へアップデートした際に後で確認しようと記載したコメントですが

- 以降の修正でgithub actionsも全てパスできた
- 商品レビュープラグインの4.3対応でも、Entity拡張でのテーブル追加等、問題はなさそう

ので、特に影響なしと判断しています。

今では不要なコメントなので削除しておきます。

**Review by nanasess — DISMISSED** [review]

**Comment by dotani1111** [comment]
2月末ぐらいを目処に、mergeしていきたいと思います。

**Review by nanasess — DISMISSED** [review]

**Review by dotani1111 — DISMISSED** [review]

**Review by dotani1111 — APPROVED** [review]

**Review by shinya — APPROVED** [review]

**Review by nanasess — APPROVED** [review]
LGTM

**Review by ji-eunsoo — APPROVED** [review]

**Pull request merged**